### PR TITLE
CBD-5063: Fixes for Arm packaging, add deb dependencies

### PR DESCRIPTION
**Pull request by udkyo**
CBD-5063

Describe your PR here...
- Add missing dependencies
- Platform introspection fixes
- Some tidying up of package-deb.rb

## Pre-review checklist
- [X] Removed debug logging (`fmt.Print`, `log.Print`, ...)
- [N/A] Logging sensitive data? Make sure it's tagged (e.g. `base.UD(docID)`, `base.MD(dbName)`)
- [N/A] Updated relevant information in the API specifications (such as endpoint descriptions, schemas, ...) in `docs/api`

## Dependencies (if applicable)
- [N/A] Link upstream PRs
- [N/A] Update Go module dependencies when merged

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [ ] `xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/000/

### CODE REVIEW

File: service/sync_gateway_service_install.sh [hunk 1]
```diff
@@ -47,20 +47,22 @@ usage() {
...
-    VER=$(cat /etc/debian_version)
+  elif [ -f /etc/os-release ]; then
+    . /etc/os-release
+    OS=$(echo "${ID}" | sed "s/.*/\u&/")
+    if [ "${OS}" = "Rhel" ]; then
+      OS=RedHat
+    elif [ "${OS}" = "Debian" ]; then
+      VER=$(cat /etc/debian_version)
+    else
+      VER=$VERSION_ID
```
Comment (mhocouchbase) [inline service/sync_gateway_service_install.sh:61]: where does VERSION_ID come from?

File: service/sync_gateway_service_uninstall.sh [hunk 2]
```diff
@@ -36,20 +36,22 @@ usage() {
...
-    VER=$(cat /etc/debian_version)
+  elif [ -f /etc/os-release ]; then
+    . /etc/os-release
+    OS=$(echo "${ID}" | sed "s/.*/\u&/")
+    if [ "${OS}" = "Rhel" ]; then
+      OS=RedHat
+    elif [ "${OS}" = "Debian" ]; then
+      VER=$(cat /etc/debian_version)
+    else
+      VER=$VERSION_ID
```
Comment (mhocouchbase) [inline service/sync_gateway_service_uninstall.sh:50]: where does VERSION_ID come from?

File: service/sync_gateway_service_upgrade.sh [hunk 3]
```diff
@@ -36,20 +36,22 @@ usage() {
...
-    VER=$(cat /etc/debian_version)
+  elif [ -f /etc/os-release ]; then
+    . /etc/os-release
+    OS=$(echo "${ID}" | sed "s/.*/\u&/")
+    if [ "${OS}" = "Rhel" ]; then
+      OS=RedHat
+    elif [ "${OS}" = "Debian" ]; then
+      VER=$(cat /etc/debian_version)
+    else
+      VER=$VERSION_ID
```
Comment (mhocouchbase) [inline service/sync_gateway_service_upgrade.sh:50]: where does VERSION_ID come from?

**Comment by udkyo** [comment]
Hi Ming, it's not entirely obvious what's happening there but VERSION_ID is a field in /etc/os-release which is sourced a few lines above each of the occurrences.

**Review by mhocouchbase — APPROVED** [review]

**Comment by bbrks** [comment]
@mhocouchbase @udkyo Let me know if you two are happy for this to be merged.

Our Jenkins doesn't run builds from forked repos for security reasons, but I can manually override the checks

**Pull request merged**